main_test.go

```diff
@@ -0,0 +1,377 @@
+		assert.Equal(t, []string(nil), opts.Patches)
+
+	})
+
+	t.Run("Verify functionality of newArgParser", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		patchFileName := command.args[2]
+		assert.Equal(t, true, opts.Linting)
+		assert.Equal(t, []string{patchFileName}, opts.Patches)
+		require.NoError(t, err)
+
+	})
+
+}
+
+func TestLoadPatches(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify correct functionality of LoadPatches", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		require.NoError(t, err)
+		_, err = loadPatches(command.fset, opts, os.Stdin)
+		require.NoError(t, err)
+	})
+	t.Run("Verify correct error gets thrown when loadPatches Failure", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs([]string{"-l", "-p",
+			"testdata/patch/error1.patch", "testdata/test_files/test2.go"})
+		require.NoError(t, err)
+		path := "testdata/patch/error1.patch"
+		_, err = loadPatches(command.fset, opts, os.Stdin)
+		expectedErrorString := "could not read \"" + path + "\": " + "open " + path + ": no such file or directory"
+		require.EqualError(t, err, expectedErrorString)
+	})
+}
+
+func TestApply(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify functionality of Apply when successful", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		filename := "testdata/test_files/lint_example/test2.go"
+		content, err := ioutil.ReadFile(filename)
+		fast, err := parser.ParseFile(command.fset, filename, content /* src */, parser.AllErrors|parser.ParseComments)
+		progs, err := loadPatches(command.fset, opts, os.Stdin)
+		patchRunner := newPatchRunner(command.fset, progs)
+		_, actualComments, _ := patchRunner.Apply(filename, fast)
+		expectedComments := []string{" This patch replaces instances of fmt.Sprintf()",
+			" with fmt.Errorf()", " Patch files can be applied to mutiple files"}
+		require.NoError(t, err)
+		assert.Equal(t, expectedComments, actualComments)
+	})
+	t.Run("Verify Apply functionality when patch doesn't apply", func(t *testing.T) {
+		_, opts := newArgParser()
+		filename := "testdata/test_files/lint_example/time.go"
+		content, _ := ioutil.ReadFile(filename)
+		fast, _ := parser.ParseFile(command.fset, filename, content /* src */, parser.AllErrors|parser.ParseComments)
+		progs, _ := loadPatches(command.fset, opts, os.Stdin)
+		patchRunner := newPatchRunner(command.fset, progs)
+		_, actualComments, ok := patchRunner.Apply(filename, fast)
+		expectedComments := []string(nil)
+		assert.Equal(t, expectedComments, actualComments)
+		assert.Equal(t, false, ok)
+	})
+
+}
+
+func TestPreviewer(t *testing.T) {
+	t.Parallel()
+	t.Run("run functionality with -l", func(t *testing.T) {
+		var tests = []struct {
+			name           string
+			give           []string
+			want           string
+			targetFilename string
```

Comment:
Just "filename" should be good enough here since there's no other "filename" to disambiguate against by adding the "target" prefix.